wallet-new/integration/Util.hs

```diff
@@ -87,6 +101,11 @@ genesisRef :: WalletRef
 genesisRef = unsafePerformIO newEmptyMVar
 {-# NOINLINE genesisRef #-}
 
+shouldFailWith :: (Show a) => Either ClientError (WalletResponse a) -> ClientError -> IO ()
+shouldFailWith eresp wantErr = do
+    gotErr <- eresp `shouldPrism` _Left
+    gotErr `shouldBe` wantErr
```

Comment:
This would actually give better test failure messages with ``eresp `shouldBe` Left wantErr``